### ISSUE TITLE
Output Directory for `exportDatatables` Function in Gradle Plugin

### ISSUE THREAD

**Issue by timtebeek** [openrewrite/rewrite-gradle-plugin#341]
### Discussed in https://github.com/openrewrite/rewrite/discussions/4603

<div type='discussions-op-text'>

<sup>Originally posted by **yusasak1** October 23, 2024</sup>
Hi,

I have questions about exportDatatables function when using Gradle plugin.

1. When setExportDatatables is set to `true`, the CSV files are output to `<repository top directory>/build/datatables`. I expected them to be output to `<gradle project top directory>/build/datatables`. Is there a reason why they must be output to the top directory of the repository?

2. In a multi-project setup with Gradle, the DataTables from the OpenRewrite tasks executed for subprojects are also output to `<repository top directory>/build/datatables`. I expected them to be output to the `build/datatables` directory of each subproject. Is there a reason why they are not output to each subproject's directory? (For example, is it for the same reason that files output with the `--profile` option in Gradle are output to the top directory of Gradle, regardless of the subproject?)

I'm using: 

- Gradle 8.10
- OpenRewrite 2.20.0
- OpenRewrite Gradle plugin 6.24.0

```terminal
<Repository Top Directory>
+---build
|   \---rewrite
|       \---datatables // Currennt export Directory
|
+---GradleTop
|   +---.gradle
|   +---app
|   |   +---build
|   |   |   +---classes
|   |   |   |   \---java
|   |   |   |       +---main
|   |   |   |       \---test
|   |   |   +---distributions
|   |   |   +---generated
|   |   |   +---libs
|   |   |   +---reports
|   |   |   |   +---rewrite
|   |   |   |   \---tests
|   |   |   +---rewrite
|   |   |       \---datatables // or should be here?
|   |   |   +---scripts
|   |   |   +---test-results
|   |   |   \---tmp
|   |   \---src
|   |
|   +---build
|   |   \---rewrite
|   |       \---datatables // should be here?
|   |
|   +---buildSrc
|   |   +---.gradle
|   |   +---build
|   |   \---src
|   |
|   +---gradle
|   |   \---wrapper
|   \---utilities
|       +---build
|       |   +---classes
|       |   |   \---java
|       |   |       \---main
|       |   +---generated
|       |   +---libs
|       |   \---tmp
|       \---src
|           +---main
|           |   +---java
|           |   \---resources
|           \---test
|               \---resources
+---otherDirA
\---otherDirB
```

Thank you.</div>

**Comment by timtebeek** [comment]
Thansk again for reporting your findings @yusasak1 . As indicated on https://github.com/openrewrite/rewrite/discussions/4603#discussioncomment-11027092 the export of data tables was contributed in 
- https://github.com/openrewrite/rewrite-gradle-plugin/pull/310

From what you've described the way the output folder is determined here can be improved
https://github.com/openrewrite/rewrite-gradle-plugin/blob/95501ff78e21fb79f73c23dbc985f957cec101fb/plugin/src/main/java/org/openrewrite/gradle/isolated/DefaultProjectParser.java#L1397-L1402

Would you be open to explore a fix on a draft PR? That would make it easier for us to review, approve and merge.

**Comment by yusasak1** [comment]
Hi @timtebeek,

I initially planned to change the directory to `build/rewrite/datatables` path of each Gradle project when we started the discussion(openrewrite/rewrite#4603). However, considering that the `rewriteDryRun` task outputs the `rewrite.patch` patch file to `<gradle project directory>/build/reports/rewrite/rewrite.patch`, I decided to change the output directory to `<gradle project directory>/build/reports/rewrite/datatables`. (Would it be better to avoid making significant changes to the directory structure?)

Thank you.
